pkg/rbac/clusterScopedAuthorization.go

```diff
@@ -0,0 +1,73 @@
+package rbac
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/doug-martin/goqu/v9"
+	"github.com/stolostron/search-v2-api/pkg/config"
+	"k8s.io/klog/v2"
+)
+
+type sharedList struct {
+	err       error
+	resources map[string][]string
+	updatedAt time.Time
+}
+
+func (cache *Cache) checkUserResources(uid string) (bool, error) {
+
+	//check if we already have the resources in cache
+	cr, resourcesExist := cache.shared[string(uid)]
```

Comment:
Is this uid the user's UID? We shouldn't need it to access shared cache resources, but we'll need it for the cached user data.